packages/ix-player/src/index.ts

```diff
@@ -383,35 +396,79 @@ class IxPlayerElement extends VideoApiElement {
     this.media?.addEventListener('emptied', updateLiveWindow);
   }
 
+  #handleRetry(callback?: (retries: number) => void) {
+    const isLoading = this.mediaLoadingIndicator?.hasAttribute('is-loading');
+    const reloadSrc = () => {
+      const { src } = this;
+      if (src !== undefined) {
+        this.src = '';
+        this.src = src;
+      }
+    };
+
+    if (this.retries > 3) {
+      this.mediaLoadingIndicator?.removeAttribute('is-loading');
+      this.mediaController?.setAttribute('media-paused', '');
+      if (callback) callback(this.retries);
+      return;
+    } else if (!isLoading) {
+      this.mediaLoadingIndicator?.setAttribute('is-loading', '');
+      this.mediaLoadingIndicator?.removeAttribute('media-paused');
+      this.mediaController?.removeAttribute('media-paused');
+      this.mediaController?.setAttribute('is-loading', '');
+    }
+
+    const retryDelay = [2000, 5000, 10000, 20000][this.retries];
+
+    setTimeout(() => {
+      this.retries += 1;
+      reloadSrc();
```

Comment:
question: I'm not sure I understand the concern, what part of this logic will not work? Can you expand on this a bit?

Reply:
you'll want to add preload none attribute to see that retries will no longer happen

Reply:
**comment**:

Got it, thanks!

I don't think this behavior is because of state resetting but rather the `#render()` method being called again. On observed attribute changes, the `initialState` will get written into the current value of `this.#state`. The initial state, purposefully, doesn't hold a value for `retries`.

```js
    if (shouldClearState && oldValue !== newValue) {
      this.#state = { ...this.#state, ...initialState };
    }
```

 If you log out the value of `state.retries` you'll see it does increment between manual clicks to the play button (despite preload being `"none"`).

I think this issue might have more to do with how a media element behaves when `preload="none"`. 

Currently, a change to the `this.src` attribute will cause a re-render. Even if `retires` is still less than greater than 0 and less than 3, the video does not try to load the`src`. This is _intended_ behavior, AFAIK, of `preload="none"`. As far as the video element is concerned, it should not play the attached media yet, it just mounted for the very first time.

**suggestion**:

What needs to happen I think is to use the `this.media?.load()` call instead of setting `this.src`. This _won't_ cause a re-render. Crucially, it should be followed by `this.media?.play()`, otherwise there will be no `onerror` event to retry on.

This way, the video element lifecycle isn't reset, and therefore `preload="none"` won't prevent the reload `src` from playing.

```js
    const reloadSrc = () => {
      const { src } = this;
      if (src !== undefined) {
        this.media?.load();
        this.media?.play();
      }
    };
// ... later
    setTimeout(() => {
      this.retries += 1;
      reloadSrc();
      if (callback) callback(this.retries);
    }, retryDelay);
  }
```

One alternative is to listen for the suspend event instead of the error event; this way, you can avoid needing to `.play()` after `.load()` the source. The trouble is you lose the context of _why_ the media was suspended. You could throw a custom error but... things start to get complicated.

**note**: The route of using `.load()` followed by `.play()` had been explored but I initially decided against it since I wasn't sure users would want the video to _play_ vs become _ready to play_.

**discussion**: Does it make sense for the video to auto-play when ready **or** should it load the media but _not_ autoplay? If we don't want it to autoplay, does it matter that `preload="none"` won't retry unless you click play? IMO the current behavior for `preload="none"` isn't ideal but the alternative (autoplay) could be worse.

**question**: is this behavior blocking or something we can follow up on?